File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
  def initialize x
    @max = x
  end

  def square_of_sums
    total = 0
    for i in 1..@max
      total += i
    end
    total**2
  end

  def sum_of_squares
    total = 0
    for i in 1..@max
      total += i**2
    end
    total
  end

  def difference
    square_of_sums - sum_of_squares
  end
end

File: python/difference-of-squares/difference_of_squares.py
def difference(num):
	return square_of_sum(num) - sum_of_squares(num)

def square_of_sum(num):
	return reduce(lambda x,y: x+y, range(1,num+1))**2

def sum_of_squares(num):
	return reduce(lambda x,y: x+(y**2), range(1,num+1))